tests/unit/phpDocumentor/Descriptor/Builder/Reflector/ClassAssemblerTest.php

```diff
@@ -140,37 +140,35 @@ protected function getClassReflectorDescriptor(?Fqsen $classFqsen = null, ?Fqsen
     /**
      * Create a descriptor builder mock
      */
-    protected function getProjectDescriptorBuilderMock() : MockInterface
+    protected function getProjectDescriptorBuilderMock() : ObjectProphecy
     {
-        $projectDescriptorBuilderMock = m::mock(ProjectDescriptorBuilder::class);
-        $projectDescriptorBuilderMock->shouldReceive('getDefaultPackage')->andReturn('\\');
-        $projectDescriptorBuilderMock->shouldReceive('buildDescriptor')->andReturnUsing(
-            static function ($param) {
-                $mock = null;
-
-                switch ($param) {
-                    case 'Properties':
-                        $mock = m::mock('phpDocumentor\Descriptor\PropertiesDescriptor');
-                        $mock->shouldReceive('getName')->once()->andReturn('Mock');
-                        $mock->shouldReceive('setParent')->once()->andReturn();
-                        break;
-
-                    case 'Method':
-                        $mock = m::mock(MethodDescriptor::class);
-                        $mock->shouldReceive('getName')->once()->andReturn('Mock');
-                        $mock->shouldReceive('setParent')->once()->andReturn();
-                        break;
-
-                    case 'Constant':
-                        $mock = m::mock(ConstantDescriptor::class);
-                        $mock->shouldReceive('getName')->once()->andReturn('Mock');
-                        $mock->shouldReceive('setParent')->once()->andReturn();
-                        break;
-                }
-
-                return $mock;
+        $projectDescriptorBuilderMock = $this->prophesize(ProjectDescriptorBuilder::class);
+        $projectDescriptorBuilderMock->getDefaultPackage()->shouldBeCalled()->willReturn('\\');
+        $projectDescriptorBuilderMock->buildDescriptor(Argument::any(), Argument::any())->will(function ($param) {
+            switch ($param) {
+                case 'Properties':
+                    $mock = $this->prophesize('phpDocumentor\Descriptor\PropertiesDescriptor');
```

Comment:
While we are here, Could you please update this to a class ref. `::class`

Reply:
Such class does not exists anymore so I've deleted the whole switch `case`.